<launch>
	<!-- lauch openni node -->
  	<include file="$(find openni_launch)/launch/openni.launch"/> 
	
	<!-- topic where the points cloud come from -->
	<arg name="tabletop_segmentation_points_in" default="/camera/depth_registered/points" />

	<!-- lauch tabletop_object_detector node -->
  	<node pkg="tabletop_object_detector" name="tabletop_segmentation" type="tabletop_segmentation" respawn="true" output="screen">
   		<!--topic remapping-->
    		<remap from="cloud_in" to="$(arg tabletop_segmentation_points_in)" />
    		<remap from="segmentation_srv" to="tabletop_segmentation" />
    		<remap from="markers_out" to="tabletop_segmentation_markers" />
    		<!-- processing and filtering frame -->
    		<!-- remove this entire block and processing and filtering will -->
    		<!-- be done with default settings, in the incoming cloud frame -->

    		<param name="processing_frame" value="/camera_rgb_optical_frame" />
    		<param name="up_direction" value="-1.0" />
    		<param name="z_filter_min" value="0.5" />
    		<param name="z_filter_max" value="0.8" />
    		<param name="y_filter_min" value="-0.1" />
    		<param name="y_filter_max" value="0.1" />
    		<param name="x_filter_min" value="-0.2" />
    		<param name="x_filter_max" value="0.2" />
    		<param name="table_z_filter_min" value="0.01" />
  		<param name="table_z_filter_max" value="0.30" /> 
		<param name="clustering_voxel_size" value="0.002" /> 
 		<param name="inlier_threshold" value="500" />
		<param name="plane_detection_voxel_size" value="0.002" />
		<param name="cluster_distance" value="0.1" />
		<param name="min_cluster_size" value="60" />
	</node>
</launch>

